gnocchi/amqp1d.py

```diff
@@ -0,0 +1,251 @@
+                            uuid.uuid4(),
+                            self.conf.amqp1d.creator,
+                            archive_policy_name=ap_name,
+                            name=metric_name,
+                            resource_id=resource.id)
+                    self.incoming.add_measures(metric, (measures[name],))
+                except Exception as exception:
+                    LOG.error("Unable to add measure %s: %s",
+                              metric_name, exception)
+
+        self.reset()
+
+    def _ensure_resource_exists(self, host_id, host):
+        try:
+            self.indexer.create_resource(
+                self.conf.amqp1d.resource_name,
+                utils.ResourceUUID(
+                    host_id,
+                    self.conf.amqp1d.creator),
+                self.conf.amqp1d.creator,
+                original_resource_id=host_id,
+                host=host)
+        except indexer.ResourceAlreadyExists:
+            pass
+            # LOG.debug("Resource %s already exists", host_id)
+        else:
+            LOG.info("Created resource for %s", host)
+
+    def _ensure_resource_type_exists(self):
+        try:
+            self.resource_type = self.indexer.get_resource_type(
+                self.conf.amqp1d.resource_name)
+        except indexer.NoSuchResourceType:
+            pass
+        if self.resource_type is None:
+            try:
+                mgr = self.indexer.get_resource_type_schema()
+                rtype = mgr.resource_type_from_dict(
+                    self.conf.amqp1d.resource_name, {
+                        "host": {"type": "string", "required": True,
+                                 "min_length": 0, "max_length": 255},
+                    }, "creating")
+                self.indexer.create_resource_type(rtype)
+            except indexer.ResourceTypeAlreadyExists:
+                LOG.debug("Resource type %s already exists",
+                          self.conf.amqp1d.resource_name)
+            else:
+                LOG.info(
+                    "Created resource type %s",
+                    self.conf.amqp1d.resource_name)
+                self.resource_type = self.indexer.get_resource_type(
+                    self.conf.amqp1d.resource_name)
+        else:
+            LOG.info(
+                "Found resource type %s",
+                self.conf.amqp1d.resource_name)
+
+    def _get_archive_policy_name(self, metric_name):
+        if self.conf.amqp1d.archive_policy_name:
+            return self.conf.amqp1d.archive_policy_name
+        # NOTE(sileht): We didn't catch NoArchivePolicyRuleMatch to log it
+        archive_policy = self.indexer.get_archive_policy_for_metric(
+            metric_name)
+        return archive_policy.name
+
+
+class AMQP1Server(MessagingHandler):
+    def __init__(self, conf, stats):
+        super(AMQP1Server, self).__init__()
+        self.stats = stats
+        self.url = conf.amqp1d.host + ":" + \
```

Comment:
if you want an URL, maybe just ask for an URL in the config directly?
We use URL in a lot of different places already.